test/src/test/java/hudson/tasks/LogRotatorTest.java

```diff
@@ -96,6 +101,17 @@ public void successVsFailureWithRemoveLastBuild() throws Exception {
         assertEquals(2, numberOf(project.getLastFailedBuild()));
     }
 
+    @Test
+    public void ableToDeleteCurrentBuild() throws Exception {
```

Comment:
Hmm, this was to check that the switch to `isLogUpdated` in `LogRotator.shouldKeepRun` did not break anything, since previously the call in `Run.run` would not have been able to delete the build with the switch. I guess though because of the redundant call via `GlobalBuildDiscarderListener` it would have worked anyway, there would just have been a log warning from the deleted code in `Run.run`. So maybe this test is no longer checking anything useful.

Reply:
I guess, in principle, someone could want to keep zero builds if they were using publishers (GitHub Checks, Slack, etc.) to track failures?